File: 23.cs
using System;
using System.Collections.Generic;

class Test {
  static void Main() {
    List<string> names = new List<string>();
    names.Add("Liz");
    names.Add("Martha");
    names.Add("Beth");
    for(int i = 0; i < names.Count; i++){
      string s = names[i];
      names[i] = s.ToUpper();
      Console.WriteLine(names[i]);
    }
  }
}

File: 23.cpp
#include <iostream>
#include <list>
#include <vector>
using namespace std;

// issue: list library (operator [])
// issue: string library (ToUpper)
int main(){
//  list<string> *names = new list<string>();
  vector<string> *names = new vector<string>();
  names->push_back("Liz");
  names->push_back("Martha");
  names->push_back("Beth");

  locale loc;
  for(int i = 0; i < names->size(); i++){
    string s = (*names)[i];
    for (string::iterator p = s.begin(); s.end() != p; ++p)
      *p = toupper(*p);
    (*names)[i] = s;
    cout << (*names)[i] << endl;
  }
}

